### PR TITLE
Made progess on Location feed

### PR DESCRIPTION
**Pull request by sabingaire**
I have made some progress on Location feed by making functions which stores and fetches the location in Datastore.
Now when the user submits the location from location-from.html, they will be redirected to a page locationfeed.html which shows all the locations. It looks like this [
Screen Shot 2019-04-16 at 4 00 37 PM

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -65,6 +65,25 @@ public void storeLocation(Location location) {
         datastore.put(locationEntity);
     }
 
+
```
Comment (elaineliucs) [inline src/main/java/com/google/codeu/data/Datastore.java]: Please consider removing this extra line.

File: src/main/java/com/google/codeu/servlets/LocationsServlet.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/locations")
+public class LocationsServlet extends HttpServlet {
+
+    private Datastore datastore;
+
+
```
Comment (elaineliucs) [inline src/main/java/com/google/codeu/servlets/LocationsServlet.java]: Please consider removing an extra line.

File: src/main/webapp/js/feed-loader.js [hunk 3]
```diff
@@ -44,6 +44,58 @@
    return messageDiv;
   }
 
+  function buildUILocation(){
```
Comment (elaineliucs) [inline src/main/webapp/js/feed-loader.js]: Consider keeping indentations consistent.

File: src/main/webapp/js/feed-loader.js [hunk 4]
```diff
@@ -44,6 +44,58 @@
...
+    fetchLocations();
+  }
+
+  // Fetch messages and add them to the page.
+   function fetchLocations(){
```
Comment (elaineliucs) [inline src/main/webapp/js/feed-loader.js]: Consider keeping indentations consistent.

File: src/main/webapp/js/feed-loader.js [hunk 5]
```diff
@@ -44,6 +44,58 @@
...
+        messageContainer.innerHTML = '';
+       }
+       locations.forEach((location) => {
+        const messageDiv = buildLocationDiv(location);
+        messageContainer.appendChild(messageDiv);
+       });
+     });
+   }
+
+     function buildLocationDiv(location){
```
Comment (elaineliucs) [inline src/main/webapp/js/feed-loader.js]: Here again as well with the indentation.

File: src/main/java/com/google/codeu/servlets/LocationsServlet.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import com.google.gson.Gson;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/locations")
```
Comment (elaineliucs) [inline src/main/java/com/google/codeu/servlets/LocationsServlet.java]: I believe Allan originally made it "/location" not "/locations." I think it's best to keep it as "/location."

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 7]
```diff
@@ -90,10 +109,10 @@ public void storeLocation(Location location) {
   }
 
   /**
-   * Fetches the messages for all users.
+   * Fetches the location for all users.
```
Comment (jonathantanmy) [inline src/main/java/com/google/codeu/data/Datastore.java]: Why this change? The method is still about getting messages, not locations.

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (jonathantanmy) [reply src/main/java/com/google/codeu/data/Datastore.java]: Just to clarify: when I am strict about extra lines, it's when you added or removed a line in an unrelated part of the code. Here, it's related - right below, you're adding a getLocations method.

Having said that, it's not necessary to have 2 lines in between methods, so feel free to remove this.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 8]
```diff
@@ -103,7 +122,6 @@ public void storeLocation(Location location) {
     PreparedQuery results = datastore.prepare(query);
 
     for (Entity entity : results.asIterable()) {
-
```
Comment (jonathantanmy) [inline src/main/java/com/google/codeu/data/Datastore.java]: This line, for example, should be re-added. You're not modifying any nearby line at all.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 9]
```diff
@@ -119,9 +137,26 @@ public int getTotalMessageCount(){
...
+      System.err.println("Error reading location.");
+      System.err.println(entity.toString());
+      e.printStackTrace();
+    }
+  }
+
   /* Helper function that encapsulates the redundant segments of the getMessages
    and getAllMessages functions; reads Messages.
    */
+
```
Comment (jonathantanmy) [inline src/main/java/com/google/codeu/data/Datastore.java]: And this line should be removed, because you're not changing any adjacent line.

File: src/main/java/com/google/codeu/servlets/LocationsServlet.java (hunk 6 above)
Comment (jonathantanmy) [reply src/main/java/com/google/codeu/servlets/LocationsServlet.java]: Yes, and also modify the LocationServlet, not create a new class.

**Review by jonathantanmy — COMMENTED** [review]
Great job :-) and thanks for including a screenshot as well.

**Review by jonathantanmy — APPROVED** [review]
